backend/server.js

```diff
@@ -0,0 +1,393 @@
+  const accessToken = req.header('Authorization')
+
+  try {
+    const user = await User.findOne({ accessToken })
+    if (user) {
+      next()
+    } else {
+      res.status(401).json({
+        response: {
+          message: 'Please, log in',
+        },
+        success: false
+      })
+    }
+  } catch (error) {
+    res.status(400).json({ response: error, success: false })
+  }
+}
+
+// Home
+app.get('/', (req, res) => {
+  res.send({ 
+    Welcome: 'Welcome to the Pomodoro-API',
+    Contributers: 'Birgit Nehrwein, Darya Lapata, Rebecca Philipson',
+    Endpoints: 'https://final-project-pomodoro-api.herokuapp.com/endpoints',
+    Documentation: 'https://documenter.getpostman.com/view/18068162/UVkiSdnR'
+  })
+})
+
+// all endpoints
+app.get('/endpoints', (req, res) => {
+  res.json({
+    response: listEndpoints(app),
+    success: true
+  })  
+})
+
+// endpoint for getting all the tasks and pomodoros of a user
+app.get('/tasks/:userId', authenticateUser)
+app.get('/tasks/:userId', async (req, res) => {
+  const { userId } = req.params
+
+  const tasks = await Task.find({ user: userId }).sort({ createdAt: 'desc' })
+  const pomodoros = await Pomodoro.find({ user: userId })
+  res.status(201).json({ response: { tasks, pomodoros}, success: true })
+})
+
+// endpoint for posting a new task
+app.post('/tasks', authenticateUser)
+app.post('/tasks', async (req, res) => {
+  const { description, user } = req.body
+
+  try {
+    const queriedUser = await User.findById(user)
+    const newTask = await new Task ({ description, user: queriedUser._id}).save()
+
+    res.status(201).json({ 
+      response: {
+        _id: newTask._id,
+        description: newTask.description, 
+        completed: newTask.completed,
+        pomodoros: newTask.pomodoros,
+        createdAt: newTask.createdAt
+      }, 
+      success: true 
+    })
+  } catch (error) {
+    res.status(400).json({ response: error, success: false })
+  }
+})
+
+// endpoint to complete existing tasks
+app.patch('/tasks/:taskId/complete', authenticateUser)
+app.patch('/tasks/:taskId/complete', async (req, res) => {
+  const { taskId } = req.params
+  const { user, completed, completedAt } = req.body
+
+  try {
+    const queriedUser = await User.findById(user)
+    const completedTask = await Task.findOneAndUpdate(
+      { _id: taskId, user: queriedUser._id},
+      { completed, completedAt },
+      { new: true }
+    )
+
+    if (!completedTask) {
+      res.status(404).json({ response: 'No task found with this Id', success: false})
+    } else {
+      res.status(200).json({ response: completedTask, success: true})
+    }
+  } catch (error) {
+    res.status(400).json({ response: error, success: false })
+  }
+})
+
+// endpoint to update the description of an existing task. 
+// The setup of the endpoint prevents changing the description of already completed tasks
+app.patch('/tasks/:taskId/update', authenticateUser)
+app.patch('/tasks/:taskId/update', async (req, res) => {
```

Comment:
Shorthand version of injecting authenticateUser middleware would be :

```
app.patch('/tasks/:taskId/update', authenticateUser, async (req, res) => { ... });
```

Reply:
Great tip!